### PR TITLE
Adapt LSP request handlers for neovim/neovim#15504.

### PR DESCRIPTION
**Pull request by rish987**

### CODE REVIEW

File: lua/lean/_util.lua [hunk 1]
```diff
@@ -50,4 +50,33 @@ function M.subprocess_check_output(opts, timeout)
...
 end
 
+function M.wrap_handler(other_handler, handler)
+  return function(...)
+    other_handler(...)
+    handler(...)
+  end
+end
+
+function M.mk_handler(fn)
```
Comment (Julian) [inline lua/lean/_util.lua:60]: This looks like it's @mfussenegger's code from https://github.com/mfussenegger/nvim-lsp-compl/commit/29a81f3747373f42d002775b1f79dcd19fada2e4 right? If so it's GPL 3.0 so we at very least have to notate that here (unless perchance @mfussenegger has already said somewhere it's OK to crimp regardless?)

**Review by Julian — APPROVED** [review]
Small license issue but after clarifying that otherwise think this is great thanks for knocking it out!

File: lua/lean/_util.lua (hunk 1 above)
Comment (rish987) [reply lua/lean/_util.lua:61]: Yep just copied it -- added acknowledgements, let me know it that is sufficient

Comment (mfussenegger) [reply lua/lean/_util.lua:61]: Fine for me to copy it.

Comment (Julian) [reply lua/lean/_util.lua:61]: Thank you (both)!

**Pull request merged**